src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.Prerendering.cs

```diff
@@ -88,6 +90,9 @@ public static void MarkAsAllowingEnhancedNavigation(HttpContext context)
         ParameterView parameters)
         => PrerenderComponentAsync(httpContext, componentType, prerenderMode, parameters, waitForQuiescence: true);
 
+    // We do not want the debugger to consider NavigationExceptions caught by this method as user unhandled.
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    [DebuggerDisableUserUnhandledExceptions]
```

Comment:
Two possible ideas on how to deal with this:
1. I am guessing this is too late for .NET 9, but I have always thought there should be a `Debugger.Break` overload that took an exception as an argument so that the target process could force a debugger to stop.
2. If it would make sense to update the linked code to flow the original navigation exception as an inner exception, maybe we could have the debugger check if the inner exception traveled through user code.

Reply:
I did flow the `NavigationException` as an inner exception to the `InvalidOperationException`. I think it makes sense to include whether or not the debugger checks for it, but that would be cool.